### PR TITLE
iox-#743 Set documentation links to `latest`/`master`

### PR DESCRIPTION
**Pull request by mossmaurice**
Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

## Pre-Review Checklist for the PR Author

1. [x] ~~Code follows the coding style of [CONTRIBUTING.md][contributing]~~
1. [x] ~~Tests follow the [best practice for testing][testing]~~
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer
* Set documentation links to `latest`/`master`

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #743 (partly)

### CODE REVIEW

**Review by elfenpiff — APPROVED** [review]

File: iceoryx_examples/icediscovery/README.md [hunk 1]
```diff
@@ -22,9 +22,9 @@ the availability of services respectively.
 We create several publishers which offer their services on construction by
 default. For more dynamism the `cameraPublishers` offer/stop their services
 periodically. If you want more information on how to create a publisher, have a
-look at the [icehello](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icehello),
-[icedelivery](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery),
-and [iceoptions](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery)
+look at the [icehello](https://github.com/eclipse-iceoryx/iceoryx/tree/master/iceoryx_examples/icehello),
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/icediscovery/README.md:25]: Why do we link to master in the icediscovery examples?

Comment (mossmaurice) [reply iceoryx_examples/icediscovery/README.md:25]: @FerdinandSpitzschnueffler After our discussion last week, we came up with the following proposal:

* `master` branch always points to `latest` on [iceoryx.io](https://iceoryx.io) and `master` for all GitHub links
* `release_x.0` uses specific version tags which are increased before new releases are published e.g. `vx.0.1` or `vx.1.0`, etc.

Comment (FerdinandSpitzschnueffler) [reply iceoryx_examples/icediscovery/README.md:25]: Oh, I overlooked that the target branch is master...

**Comment by elBoberido** [comment]
With the backport of `release_2.0` to `master`  this PR should be obsolete

**Comment by mossmaurice** [comment]
> With the backport of `release_2.0` to `master` this PR should be obsolete

I fear that's not the case, some links are still pointing to `v2.0.0` on `master` and not to `master` / `latetest` e.g.

```
iceoryx_examples/iceoptions/README.md:100
[here](https://iceoryx.io/v2.0.0/getting-started/overview/#restrictions)

iceoryx_examples/complexdata/README.md:7
For more information about this limitation see the [QoS article](https://iceoryx.io/v2.0.0/concepts/qos-policies/).
```

**Comment by mossmaurice** [comment]
> @mossmaurice since the homepage doesn't have a master version, it is not possible use master is such cases yet

That's true, there is CI set up for that yet. However, MkDocs a [supports](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/) redirection feature already forwarding to latest version via [`iceoryx.io/latest`](https://github.com/eclipse-iceoryx/iceoryx-web/blob/main/index.html#L10). Once we have `master` on  [iceoryx.io](https://iceoryx.io) the `latest` redirection should be removed. Till then it would make more sense to me to link to `latest` than a fixed version which will forward to `master` in future, e.g. imagine someone works on on old branch clicks on `latest` link and is then redirected to `master` because the CI builds it then.

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Pull request merged**